pkg/endpointmanager/manager.go

```diff
@@ -364,6 +404,32 @@ func RegenerateAllEndpoints(owner endpoint.Owner, regenMetadata *endpoint.Extern
 	return &wg
 }
 
+// RegenerateEndpointSet is similar to RegenerateAllEndpoints, except it only
+// regenerates all endpoints with IDs in endpointIDs.
+func RegenerateEndpointSet(owner endpoint.Owner, regenMetadata *endpoint.ExternalRegenerationMetadata, endpointIDs map[uint16]struct{}) *sync.WaitGroup {
+	var wg sync.WaitGroup
+	wg.Add(len(endpointIDs))
+
+	for endpointID := range endpointIDs {
+		ep := endpoints[endpointID]
+		go func(ep *endpoint.Endpoint, wg *sync.WaitGroup) {
```

Comment:
Make this `go func` a separate function since it's exactly the same as above in `RegenerateAllEndpoints`